src/containers/Bookings.js

```diff
@@ -1,19 +1,62 @@
-import React, { Component } from 'react';
-import Search from '../components/Search.js';
-// import Results from '../components/Results.js';
-// import FakeBookings from '../data/fakeBookings.json';
+import React, { Component } from "react";
+import Search from "../components/Search.js";
+import BookingsMessage from "../components/BookingsMessage.js";
+import Results from "../components/Results.js";
+import fakeBookings from "../data/fakeBookings.json";
+import Counter from "../components/Counter";
 
 export default class Bookings extends Component {
-  search = () => {
-    console.info('TO DO!');
+  state = {
+    count: 0,
+    count2: 0,
+    sum: 0,
+    input: 0,
+    bookings: fakeBookings
   };
-  
+
+  userInput = e => {
+    e.preventDefault();
+    this.setState({
+      input: parseInt(e.target.value, 10)
+    });
+  };
+
+  bookingsById = () => {
+    this.setState({
+      bookings: fakeBookings.filter(booking => {
+        return booking.id === this.state.input;
+      })
+    });
+  };
+
+  increase = () => {
```

Comment:
what you've done here is correct in previous versions of React but there is a new recommended way in the current version. https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous (setState can take a function that has the previous state passed to it as an argument)

Reply:
Would it be like this: 
```
this.setState((prevState, props) => ({
  count: ++prevState.count,
  sum: ++prevState.sum
}));
```
And do I need to use props argument here?

Reply:
yeah it should look this